### PR TITLE
Enable simulation with rebuild

### LINKED ISSUE #886: implement an update model for models that need to be re-built. `get_requires_rebuild(model) == true`
This feature needs to be implemented as follows 

```Julia
function update_model!(model::OperationModel, sim::Simulation)
    if get_requires_rebuild(model)
        # initial_conditions_data =  make_initial_conditions_data()
        # Reset and flush of optimization container()
        # Set initial conditions in container
        # build_impl!(model.container, template, system)
    else
        update_model!(model, get_simulation_state(sim), get_ini_cond_chronology(sim))
    end
    return
end
```

Items: 

- [ ] Implement a make initial conditions data object the returns an `InitialConditionsData` populated correctly with the values for the parameters and variables. 
- [ ] Implement a method to flush the optimization problem from the optimization container. Hint: use `clear!` from Dicts. 
- [ ] Modify the implementation of the FeedForwards to check if the parameters for the FeedForwards exist in `InitialConditionsData` to initialize the values of the parameters for the FeedForwards.

### PR DESCRIPTION
**Pull request by jd-lara**
This is work in progress for enabling rebuild and close #886. 

Some notes for the pending items. 

1. if the model runs with rebuild parameters can't be VariableRefs otherwise the associated JuMP models will be flushed and the variable will be "orphan". 
2. Parameters for the FF's need to be kept too.

### CODE REVIEW

File: src/operation/operation_model_interface.jl [hunk 1]
```diff
@@ -15,12 +15,12 @@ get_name(model::OperationModel) = model.name
 get_store(model::OperationModel) = model.store
 is_synchronized(model::OperationModel) = is_synchronized(get_optimization_container(model))
 
-function get_requires_rebuild(model::OperationModel)
+function get_rebuild_model(model::OperationModel)
     sim_info = get_internal(model).simulation_info
     if sim_info === nothing
-        error("Model not part of a simulation")
+        error("Model not part of a simulation")ß
```
Comment (daniel-thom) [inline src/operation/operation_model_interface.jl:21]: Is the character at the end a typo?

Comment (jd-lara) [reply src/operation/operation_model_interface.jl:21]: I have no idea how that character arrived there

File: src/parameters/add_parameters.jl [hunk 2]
```diff
@@ -41,6 +64,9 @@ function add_parameters!(
     U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}},
     W <: AbstractDeviceFormulation,
 } where {D <: PSY.HybridSystem}
+    if get_rebuild_model(get_settings(container)) && has_container_key(container, T, D)
```
Comment (daniel-thom) [inline src/parameters/add_parameters.jl:67]: Is there documentation that explains this to model developers?

Comment (daniel-thom) [reply src/parameters/add_parameters.jl:67]: And is there a way to automate this so that it doesn't have to live everywhere?

Comment (jd-lara) [reply src/parameters/add_parameters.jl:67]: not yet :\

**Pull request merged**